integr8ly/integreatly-operator
# adds new authentication flow to RHSSO to delay user login

# Description

Add a new Authentication Flow that delays users login until they get created on 3scale by the RHMI operator and flagged as created in KeycloakUser CR. This new authentication flow will replace the First Broker login authentication flow in openshift realm of RHSSO. 

https://issues.redhat.com/browse/INTLY-6491

This PR requires other PRs to be merged and e2e test to be implemented
- keycloak client api https://github.com/integr8ly/keycloak-client/pull/7
- keycloak operator downstream https://github.com/integr8ly/keycloak-operator-rhmi/pull/7


Link to the Keycloak authentication delay plugin: https://github.com/integr8ly/authentication-delay-plugin

## Type of change




## Verification steps

1) Install the operator locally against a OSD cluster
2) Add testing IDP  `NUM_ADMIN=5 NUM_REGULAR_USER=5 ./scripts/setup-sso-idp.sh`
3) Login to 3scale with one of the Openshift users via RHSSO
4) Ensure you a redirected to a page that shows the message `Your account is being provisioned` 
5) Verify user is created in RHSSO and gets the attribute `3scale_user_created` set to `true`
6) Verify if user is redirected and authenticated in 3scale.
